src/error.js

```diff
@@ -0,0 +1,63 @@
+import constants from "./constants.js";
+
+export default class SqliteError extends Error {
+  /**
+   * SqliteError
+   *
+   * Extension over the standard JS Error object
+   * to also contain class members for error code
+   * and error code name
+   *
+   *     try {
+   *      db.query(`Syntax error`);
+   *     } catch (error) {
+   *       console.log(`Error message: ${error.message}`);
+   *       console.log(`Error code: ${error});
+   *       console.log(`Error code name: ${error.codename});
+   *     }
+   *
+   */
+
+  name = "SqliteError";
+
+  constructor(message, code) {
+    super(message);
+    this.code = code ?? null;
+  }
+
+  /**
+   * SqliteError.code
+   *
+   * The SQLite result error code,
+   * see the SQLite docs for more
+   * information about each error code.
+   *
+   * [https://www.sqlite.org/c3ref/c_abort.html](https://www.sqlite.org/c3ref/c_abort.html)
+   *
+   * Beyond the SQLite error code, this member
+   * can also contain custom error codes specific
+   * for this library:
+   *
+   *     stmtLimit:         1000
```

Comment:
Oh yes a markdown table makes more sense.

I think it would be appropriate to have all codes here, I'm fixing it for now until we decide what to do. It looks pretty bad on the generated docs site though.

Reply:
I think leaving them in is better. You can try making the table wider, by having 4 columns, that may look better. But I'd be willing to trade prettiness for completes in the api doc 😄

Reply:
Yes, sounds like a good trade-off.